gnuradio-runtime/lib/top_block_impl.cc

```diff
@@ -129,6 +132,21 @@ namespace gr {
   void
   top_block_impl::wait()
   {
+    do {
```

Comment:
You are right, the test condition is always checked even if continue is used. So this is wrong, I will rewrite and pus to the new PR.